primefaces/src/main/java/org/primefaces/model/filter/ComparableFilterConstraint.java

```diff
@@ -23,17 +23,60 @@
  */
 package org.primefaces.model.filter;
 
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.text.ParseException;
 import javax.faces.context.FacesContext;
 import java.util.Locale;
+import java.util.function.BiPredicate;
 
-public abstract class ComparableFilterConstraint extends StringFilterConstraint {
+public abstract class ComparableFilterConstraint implements FilterConstraint {
 
     @Override
     public boolean isMatching(FacesContext ctxt, Object value, Object filter, Locale locale) {
+        if (value == null) {
+            return false;
+        }
         if (!(value instanceof Comparable) || !(filter instanceof Comparable)) {
             throw new IllegalArgumentException("Invalid type: " + value.getClass() + ". Valid type: " + Comparable.class.getName());
         }
 
-        return super.isMatching(ctxt, value, filter, locale);
+        if (value instanceof Number) {
+            return getPredicateBigDecimal()
+                    .test(new BigDecimal(value.toString()),
```

Comment:
shouldnt this use the locale as well?

Reply:
No, this is the value from the model. The other string is the entered filter value and can have a decimal separator based on the locale.

Reply:
but coulnd this destroy the number somehow? Like value is Double, you do a toString and new BigDecimal on it? What about the locale in this case?

Reply:
https://stackoverflow.com/questions/16216248/convert-java-number-to-bigdecimal-best-way